### PR TITLE
Ruby Class Enhancements

### PR DESCRIPTION
**Pull request by stino-x**
This pull request includes a series of enhancements and new class implementations in Ruby. The changes made in this pull request include:

Person Class: Extended the Person class, including additional methods and refinements to the constructor. Also, added a Nameable module for correct name handling.

Student Class: Created a Student class that inherits from Person, allowing for additional methods related to students.

Classroom Class: Introduced a Classroom class with a has-many/belongs-to relationship with students.

Book Class: Created a Book class with title and author attributes.

Rental Class: Implemented a Rental class that enables a many-to-many relationship between Person and Book through rentals.

Each class and relationship has been explained in detail and designed for enhanced functionality, making it easier to manage classrooms, students, books, and rentals.

Please review and merge this pull request. Thank you!

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    student.classroom = self
+    students << student
```
Comment (Efiamotu-1) [inline classroom.rb]: - [ ] You created a variable `@students` on line 7 to be an empty array but on this highligted line 12 you are not referring to that variable, please kindly correct this line of code by useing `@students` :+1: :sparkles:

File: person.rb
Comment (Efiamotu-1) [inline person.rb:1]: - [ ] you are missing the method that serves as the 'belongs-to' side (a student belongs to a classroom). see the below code snippet below to achieve this in your student.rb file :+1: 
- code snippet
```ruby
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: book.rb
Comment (Efiamotu-1) [inline book.rb:1]: - [ ] `A method for adding rentals should be present in for every book :+1: kindly visit this [lesson](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`) to learn more about this

File: person.rb
Comment (Efiamotu-1) [inline person.rb:1]: - [ ] `A method for adding rentals should be present in for every person :+1: kindly visit this [lesson](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`) to learn more about this

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @stino-x, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb
Comment (stino-x) [reply person.rb:1]: hi @Efiamotu-1 are you sure that this goes in the person class?

Comment (Efiamotu-1) [reply person.rb:1]: @stino-x I noted that it should be put in your student.rb file

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @stino-x, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  Implemented required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**